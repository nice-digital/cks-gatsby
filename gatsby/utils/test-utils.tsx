import React, { ReactElement } from "react";
import { render, RenderResult } from "@testing-library/react";
import {
	createMemorySource,
	createHistory,
	LocationProvider,
	History,
} from "@reach/router";

// Util for rendering a component wrapped within Reach Router's LocationProvider
// See https://testing-library.com/docs/example-reach-router
export const renderWithRouter = (
	ui: ReactElement,
	{ route = "/", history = createHistory(createMemorySource(route)) } = {}
): { history: History } & RenderResult => ({
	...render(<LocationProvider history={history}>{ui}</LocationProvider>),
	// adding `history` to the returned utilities to allow us
	// to reference it in our tests (just try to avoid using
	// this to test implementation details).
	history,
});
