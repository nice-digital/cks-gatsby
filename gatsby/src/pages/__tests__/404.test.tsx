import React from "react";

import { waitFor, screen } from "@testing-library/react";
import { renderWithRouter } from "test-utils";
import NotFoundPage from "../404";

describe("404", () => {
	it("should match snapshot", () => {
		const { container } = renderWithRouter(<NotFoundPage />);

		expect(container).toMatchSnapshot();
	});

	it("should render h1 with correct text", () => {
		renderWithRouter(<NotFoundPage />);

		expect(screen.queryByText("We can't find this page")?.tagName).toBe("H1");
	});

	it("should render a noindex robots meta tag", async () => {
		renderWithRouter(<NotFoundPage />);

		await waitFor(() => {
			// eslint-disable-next-line testing-library/no-node-access
			expect(document.querySelector("meta[name='robots']")).toHaveAttribute(
				"content",
				"noindex"
			);
		});
	});
});
