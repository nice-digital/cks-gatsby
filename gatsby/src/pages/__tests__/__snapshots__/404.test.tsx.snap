// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 should match snapshot 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="https://www.nice.org.uk/"
        >
          NICE
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/"
        >
          CKS
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Page not found
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "CKS"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "name": "Page not found"
      }
    }
  ]
}
  </script>
  <div
    class="page-header"
    id="content-start"
  >
    <h1
      class="page-header__heading "
    >
      We can't find this page
    </h1>
    <p
      class="page-header__lead"
    >
      Check that the web address has been typed correctly.
    </p>
  </div>
  <p>
    You can also try:
  </p>
  <ul>
    <li>
      looking for it using the search box
    </li>
    <li>
      browsing for it from the 
      <a
        href="/"
      >
        CKS homepage
      </a>
      .
    </li>
  </ul>
  <h2>
    Contact us
  </h2>
  <p>
    <a
      href="https://www.nice.org.uk/get-involved/contact-us"
    >
      Get in touch
    </a>
     
    if you think there is a problem.
  </p>
</div>
`;
